src/modules/ibmtts.c

```diff
@@ -1698,43 +1819,86 @@ static char *ibmtts_search_for_sound_icon(const char *icon_name)
 	return fn;
 }
 
-void alloc_voice_list()
+static void alloc_voice_list()
 {
-	enum ECILanguageDialect aLanguage[MAX_NB_OF_LANGUAGES];
-	int nLanguages = MAX_NB_OF_LANGUAGES;
+	enum ECILanguageDialect aLanguage[VOX_MAX_NB_OF_LANGUAGES];
+	int nLanguages = VOX_MAX_NB_OF_LANGUAGES;
 	int i = 0;
 
-	if (eciGetAvailableLanguages(aLanguage, &nLanguages))
+	/* if voxGetVoices available, update the list of installed voices */
+	if (_voxGetVoices) {
+		number_of_voices = VOX_RESERVED_VOICES;
+		if (!_voxGetVoices(voices, &number_of_voices) && (number_of_voices <= VOX_RESERVED_VOICES)) {
+			int i, j;
+			int min_id = eciLocales[VOX_ECI_VOICES-1].langID;			
+			for (i=0, j=VOX_ECI_VOICES; i<number_of_voices; i++) {
+				eciLocale *local = eciLocales + j;
+				size_t len = 0;
+				vox_t *vox = voices + i;
+				if (vox->id <= min_id) /* id already known? */
```

Comment:
With the dynamically-allocated array, I wouldn't bother trying to eliminate the already-known langages. Possibly voxin would revamp the ids someday and that'd screw everything up. Since long-run-wise it provides a function that just gives the set of language, I'd say just use that and forget about the hardcoded information.